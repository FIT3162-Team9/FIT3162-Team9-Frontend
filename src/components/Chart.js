import React, { PureComponent } from 'react';
import './../App.css'
import {
  AreaChart, Area, XAxis, YAxis, CartesianGrid, Tooltip, ResponsiveContainer
} from 'recharts';
import moment from 'moment';
import Paper from '@material-ui/core/Paper';

function formatXAxis(tickItem) {
  return moment.unix(tickItem).format('YYYY-MM-DD')
}


export default ({ data }) => {
  
  return (
      <ResponsiveContainer height={300} width="95%">
        <AreaChart data={data} >
          <defs>
            <linearGradient id="colorUv" x1="0" y1="0" x2="0" y2="1">
              <stop offset="1%" stopColor="#FAD6A5" stopOpacity={0.8} />
              <stop offset="99%" stopColor="#FAD6A5" stopOpacity={0} />
            </linearGradient>
            <linearGradient id="colorPv" x1="0" y1="0" x2="0" y2="1">
              <stop offset="1%" stopColor="#ABD1F3" stopOpacity={0.8} />
              <stop offset="99%" stopColor="#ABD1F3" stopOpacity={0} />
            </linearGradient>
          </defs>
          <XAxis dataKey="timestamp" tickFormatter={formatXAxis} />
          <YAxis unit='Â°C' />
          <CartesianGrid strokeDasharray="3 3" />
          <Tooltip />
          <Area type="monotone" dataKey="min" stroke="#FAD6A5" fillOpacity={1} fill="url(#colorUv)" />
        </AreaChart>
      </ResponsiveContainer>
  )
}