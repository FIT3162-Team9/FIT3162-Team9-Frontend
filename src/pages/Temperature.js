import React, { Component, useState, useEffect} from 'react'
import TemperatureVisualisation from "../components/Visualisation"
import Paper from '@material-ui/core/Paper'
import Grid from '@material-ui/core/Grid'
import clsx from 'clsx'
import { makeStyles } from '@material-ui/core/styles'
import DatePicker from 'react-datepicker';
import Container from '@material-ui/core/Container';
import 'react-datepicker/dist/react-datepicker.css'

const useStyles = makeStyles(theme => ({
    paper: {
      display: 'flex',
      overflow: 'auto',
      flexDirection: 'column',
      marginRight: '0px',
    },
    autoHeight: {
      height: '350px',
      width: 'auto',
  
    },

    container: {
      paddingTop: theme.spacing(4),
      paddingBottom: theme.spacing(4),
    },
  }))

function Temperature(props) {


  useEffect(() => {
    console.log('TEMPERATURE PROPS ---- ', props.station);
  }, [props])

  //Requires attention: Year of start date should be earlier than year of end date.  

  const classes = useStyles()

  const autoHeightPaper = clsx(classes.paper, classes.autoHeight)

  return (
    <Container maxWidth="lg" className={classes.container}>
      <Grid container spacing={3}>
        <Grid item xs={12} md={12} lg={12}>
          <Paper className={autoHeightPaper}>
            <TemperatureVisualisation station={props.station}/>
          </Paper>
        </Grid>
      </Grid>
    </Container>
        )
}

export default Temperature
