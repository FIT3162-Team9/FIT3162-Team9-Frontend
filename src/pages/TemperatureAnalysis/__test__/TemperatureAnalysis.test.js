import React from 'react';
import { shallow, mount } from "enzyme";
import toJson from "enzyme-to-json";
import Chart from '../TemperatureChart'
import Temperature from '..';

it("TemperatureChart renders without crashing", () => {
  shallow(<Chart />);
});

it("TemperatureChart renders correctly", () => {
  const tree = shallow(<Chart />);
  expect(toJson(tree)).toMatchSnapshot();
});

it("Temperature renders without crashing", () => {
  shallow(<Temperature />);
});

it("Temperature renders correctly", () => {
  const tree = shallow(<Temperature />);
  expect(toJson(tree)).toMatchSnapshot();
});
