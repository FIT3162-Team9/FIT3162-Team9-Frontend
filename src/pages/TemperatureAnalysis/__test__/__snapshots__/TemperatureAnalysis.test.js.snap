// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Temperature renders correctly 1`] = `
<WithStyles(ForwardRef(Container))
  className="makeStyles-container-3"
  maxWidth="lg"
>
  <WithStyles(ForwardRef(Grid))
    container={true}
    spacing={3}
  >
    <WithStyles(ForwardRef(Grid))
      item={true}
      lg={12}
      md={12}
      xs={12}
    >
      <WithStyles(ForwardRef(Paper))
        className="makeStyles-paper-1 makeStyles-autoHeight-2"
      >
        <TemperatureVisualisation />
      </WithStyles(ForwardRef(Paper))>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
</WithStyles(ForwardRef(Container))>
`;

exports[`TemperatureChart renders correctly 1`] = `
<ResponsiveContainer
  debounce={0}
  height={300}
  width="95%"
>
  <LineChart
    barCategoryGap="10%"
    barGap={4}
    layout="horizontal"
    margin={
      Object {
        "bottom": 5,
        "left": 5,
        "right": 5,
        "top": 5,
      }
    }
    reverseStackOrder={false}
    stackOffset="none"
  >
    <XAxis
      allowDataOverflow={false}
      allowDecimals={true}
      allowDuplicatedCategory={true}
      dataKey="timestamp"
      domain={
        Array [
          0,
          "auto",
        ]
      }
      height={30}
      hide={false}
      mirror={false}
      orientation="bottom"
      padding={
        Object {
          "left": 0,
          "right": 0,
        }
      }
      reversed={false}
      scale="auto"
      tickCount={5}
      tickFormatter={[Function]}
      type="category"
      width={0}
      xAxisId={0}
    />
    <YAxis
      allowDataOverflow={false}
      allowDecimals={true}
      allowDuplicatedCategory={true}
      domain={
        Array [
          0,
          "auto",
        ]
      }
      height={0}
      hide={false}
      mirror={false}
      orientation="left"
      padding={
        Object {
          "bottom": 0,
          "top": 0,
        }
      }
      reversed={false}
      scale="auto"
      tickCount={5}
      type="number"
      unit="Â°C"
      width={60}
      yAxisId={0}
    />
    <Tooltip
      active={false}
      allowEscapeViewBox={
        Object {
          "x": false,
          "y": false,
        }
      }
      animationDuration={400}
      animationEasing="ease"
      contentStyle={Object {}}
      coordinate={
        Object {
          "x": 0,
          "y": 0,
        }
      }
      cursor={true}
      cursorStyle={Object {}}
      filterNull={true}
      isAnimationActive={true}
      itemStyle={Object {}}
      labelStyle={Object {}}
      offset={10}
      separator=" : "
      useTranslate3d={false}
      viewBox={
        Object {
          "x1": 0,
          "x2": 0,
          "y1": 0,
          "y2": 0,
        }
      }
      wrapperStyle={Object {}}
    />
    <CartesianGrid
      fill="none"
      horizontal={true}
      horizontalFill={Array []}
      horizontalPoints={Array []}
      stroke="#eee"
      strokeDasharray="5 5"
      vertical={true}
      verticalFill={Array []}
      verticalPoints={Array []}
    />
    <Line
      activeDot={true}
      animateNewValues={true}
      animationBegin={0}
      animationDuration={1500}
      animationEasing="ease"
      connectNulls={false}
      dataKey="max"
      dot={false}
      fill="#fff"
      hide={false}
      isAnimationActive={true}
      legendType="line"
      onAnimationEnd={[Function]}
      onAnimationStart={[Function]}
      points={Array []}
      stroke="#8884d8"
      strokeWidth={1}
      type="monotone"
      xAxisId={0}
      yAxisId={0}
    />
    <Line
      activeDot={true}
      animateNewValues={true}
      animationBegin={0}
      animationDuration={1500}
      animationEasing="ease"
      connectNulls={false}
      dataKey="min"
      dot={false}
      fill="#fff"
      hide={false}
      isAnimationActive={true}
      legendType="line"
      onAnimationEnd={[Function]}
      onAnimationStart={[Function]}
      points={Array []}
      stroke="#82ca9d"
      strokeWidth={1}
      type="monotone"
      xAxisId={0}
      yAxisId={0}
    />
  </LineChart>
</ResponsiveContainer>
`;
